wpdetect/__main__.py

```diff
@@ -1,154 +1,111 @@
-                                    print("\n" + str(url) + " may not be using WordPress.\n")
-
-
-def url_check(url):
-    print("Checking: " + str(url))
+import os
+import validators
+import threading
+import urllib.request
+from urllib.parse import urlparse
+
+
+
+def isUrl(string):
+    return validators.url(string)
+
+def splitArray(array, n):
+    final = [array[i * n:(i + 1) * n] for i in range((len(array) + n - 1) // n )]
+    return final
+
+
+
+def wp_check(url): # this function is defined to support the implementation of previous version of the tool
     try:
-        if url[:4] != "http":
-            print("[!] No protocol specified.")
-            url = "http://" + url
-            print("[+] Going with HTTP.\n")
-            print("Checking: " + str(url))
-        req = urllib.request.Request(url, headers={'User-Agent': header})
-        u = urllib.request.urlopen(req)
-        if url != u.geturl():
-            print("[!] " + url + " redirected to "+u.geturl())
-            url = u.geturl()
-            print("Checking: " + str(url))
-        wp_check(url)
-    except urllib.error.HTTPError as e:
-        if e.code == 403:
-            print("Got 403! Website seems to be behind a WAF.")
-    except urllib.error.URLError:
-        if url[:5] == "https":
-            print("[!] Couldn't connect over HTTPS.")
-            print("[+] Trying with HTTP.\n")
-            url = "http://" + url[8:]
-            print("Checking: " + str(url))
-            try:
-                url_check(url)
-            except urllib.error.URLError:
-                print("Couldn't open url, please make sure to type a valid and publicly accessible url.\n")
-        else:
-            print("Couldn't open url, please make sure to type a valid and publicly accessible url.\n")
-    except ValueError:
-        print("Invalid url! Please type in correct url.\n")
+        isWp = False
+        checks = ['wp-content', 'wp-includes', 'wp-json']
+        hostname = urlparse(url).netloc
+        request = urllib.request.Request(url, headers={'User-Agent': UserAgent})
+        response = urllib.request.urlopen(request).read().decode()
+        for check in checks:
+            checkStr = hostname + "/" + check
+            if checkStr in response:
+                isWp=True
+        return isWp
+    except:
+        return False
+
+
+UserAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.75.14 (KHTML, like Gecko) Version/7.0.3 Safari/7046A194A"
 
-def help():
-    print("\nSyntax: wpdetect <website-url>")
-    print("Example: wpdetect https://iamlizu.com/")
-    print("or supply a list with '-f' flag")
-    print("Example: wpdetect -f domainlist.txt\n")
 
-def main():
+def isWordpress(urls, mode, timeOut):
     try:
-        print(figlet_format('     wpdetect     '))
-        print("=================== version: 1.3.6 ===================\n")
-        if sys.argv[1] == '-h' or sys.argv[1] == '--help':
```

Comment:
Help or any other options should have a shorter flag as well along with long flag. Such as `-h` and `--help` must show help menu.